web/controllers/userController/index.ts

```diff
@@ -1007,7 +1011,66 @@ async function putActive(wix_id: string, active: boolean, person: Pupil | Studen
                 await dissolveMatch(matches[i], 0, person);
             }
 
-            // Step 2: Deactivate
+            // Step 2: Cancel all courses if user is student
+            if (type == Student) {
+                let courses = await getConnection()
+                    .getRepository(Course)
+                    .createQueryBuilder("course")
+                    .leftJoinAndSelect("course.instructors", "instructors")
+                    .getMany();
+
+                courses.forEach(async (course: Course) => {
+                    if (course.instructors.some(i => i.id === person.id)) { // Only proceed if we're part of this course as an instructor
```

Comment:
As a small stylistic note: Through "short circuiting" one can avoid having a lot of nested blocks which becomes quite hard to read, e.g. through inverting the condition `if(!course.instructors.some(it => it.id === person.id)) continue;`